pandas/tests/reshape/concat/test_concat.py

```diff
@@ -826,3 +826,17 @@ def test_concat_mismatched_keys_length():
         concat((x for x in sers), keys=(y for y in keys), axis=1)
     with tm.assert_produces_warning(FutureWarning, match=msg):
         concat((x for x in sers), keys=(y for y in keys), axis=0)
+
+
+def test_concat_datetime64_diff_resolution():
+    # GH#53640
+    df1 = DataFrame({"a": [0, 1], "b": [4, 5]}, dtype="datetime64[s]")
+    df2 = DataFrame({"a": [2, 3], "b": [6, 7]}, dtype="datetime64[ms]")
+    result = concat([df1, df2])
+
+    expected = DataFrame(
+        data={"a": [0, 1000, 2, 3], "b": [4000, 5000, 6, 7]},
+        index=[0, 1, 0, 1],
+        dtype="datetime64[ms]",
```

Comment:
i dont think its obvious that the output dtype should be dt64[ms].  if we follow find_common_dtype logic i think it should be object.  There's a compelling argument that dt64[ms] is nicer, but there are going to be cases where that causes overflows

Reply:
How do you recommend modifying this test?

Reply:
Is there any update on this @jbrockmendel? If not `datetime64[ms]` then what do you recommend it should be?

Reply:
i think we agreed on the previous dev call that this should use the same logic as arithmetic between dt64s with different resos, which upcasts to the higher resolution, raising if there's an overflow.  @MarcoGorelli does your recollection match mine?

Reply:
yup (or at least, this sounds reasonable)

Reply:
Thanks for the explantion and sorry for the late reply (I've been too busy recently). Since we should upcast to the higher resolution, isn't `datetime64[ns]` correct here?

Reply:
I think this is fine now and that the comment is out-dated - resolving then